### PR TITLE
[SPARK-46849][SQL] Run optimizer on CREATE TABLE column defaults

### PR DESCRIPTION
**Pull request by dtenedor**
### What changes were proposed in this pull request?

This PR updates Catalyst to run the optimizer over `CREATE TABLE` column default expressions. 

### Why are the changes needed?

This helps speed up future commands that assign default values within the table.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The functionality is covered by existing tests.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by dtenedor** [comment]
cc @cloud-fan

**Review by cloud-fan — APPROVED** [review]

**Review by dongjoon-hyun — COMMENTED** [review]
Please fix the two UT failures.
```
[info] - SPARK-38335: Implement parser support for DEFAULT values for columns in tables *** FAILED *** (8 milliseconds)
[info]   == FAIL: Plans do not match ===
[info]   !'CreateTable [StructField(a,IntegerType,true), StructField(b,StringType,false)], unresolvedtablespec(Some(parquet), optionlist(), None, None, None, false), false, [1=abc]   'CreateTable [StructField(a,IntegerType,true), StructField(b,StringType,false)], unresolvedtablespec(Some(parquet), optionlist(), None, None, None, false), false
[info]    +- 'UnresolvedIdentifier [my_tab], false                                                                                                                                     +- 'UnresolvedIdentifier [my_tab], false (PlanTest.scala:179)
```

```
[info] - DESCRIBE TABLE EXTENDED of a V2 table with a default column value *** FAILED *** (8 milliseconds)
[info]   java.lang.IllegalArgumentException: requirement failed: Literal must have a corresponding value to bigint, but class Integer found.
[info]   at scala.Predef$.require(Predef.scala:337)
```

**Comment by dtenedor** [comment]
> Please fix the two UT failures.
@dongjoon-hyun this is done.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala [hunk 1]
```diff
@@ -456,13 +456,16 @@ trait V2CreateTablePlan extends LogicalPlan {
...
+ * mapping 1:1 with the fields in [[tableSchema]].
  */
 case class CreateTable(
     name: LogicalPlan,
     tableSchema: StructType,
     partitioning: Seq[Transform],
     tableSpec: TableSpecBase,
-    ignoreIfExists: Boolean)
+    ignoreIfExists: Boolean,
+    defaults: Seq[Option[Expression]])
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala:468]: Could you rename `defaults` to `defaultValues` or `defaultValueExpressions` here? `default` looks a little too broad in the context of `Table` definition.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM with one minor comment about renaming.
Thank you, @dtenedor .

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala (hunk 1 above)
Comment (dtenedor) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala]: Sure, this is done.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Oh, could you take a look at the failures, @dtenedor ?
```
[info] *** 8 TESTS FAILED ***
[error] Failed: Total 10008, Failed 8, Errors 0, Passed 10000, Ignored 28
[error] Failed tests:
[error] 	org.apache.spark.sql.execution.command.v2.AlterTableDropPartitionSuite
[error] 	org.apache.spark.sql.DSV2SQLInsertTestSuite
[error] 	org.apache.spark.sql.connector.DataSourceV2SQLSuiteV1Filter
[error] 	org.apache.spark.sql.execution.command.v2.AlterTableAddPartitionSuite
[error] 	org.apache.spark.sql.execution.command.v2.AlterTableRenamePartitionSuite
```

**Comment by dtenedor** [comment]
> Oh, could you take a look at the failures, @dtenedor ?

@dongjoon-hyun this is done.

**Comment by dongjoon-hyun** [comment]
Thank you!

**Comment by dtenedor** [comment]
cc @dongjoon-hyun @cloud-fan OK this looks stable now, all CI is passing.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @dtenedor and @cloud-fan .

**Pull request closed without merging**